File: euler6.ex
defmodule Euler6 do
  @numeros Enum.to_list(1..100)

  def sumaCuadrados() do
    @numeros
    |>Stream.map(&(&1 * &1))
    |>Enum.sum
  end

  def cuadradoSuma() do
    @numeros
    |>Enum.sum
    |>:math.pow(2)
    |>round
  end

  def resultado do
    cuadradoSuma() - sumaCuadrados
  end
end

File: euler6.c
#include <stdio.h>
#include <time.h>
#include <math.h>

long sumaCuadrados();
long cuadradoSuma();

int main() {
	long sumaCua,cuaSuma;
	
	clock_t begin, end;
	double time_spent;
	
	begin = clock();
	
	sumaCua = sumaCuadrados();
	cuaSuma = cuadradoSuma();
	
	end = clock();
	time_spent = (double)(end-begin) / CLOCKS_PER_SEC;
	
	printf("\n Resultado:%ld", cuaSuma-sumaCua);
	printf("\nTiempo de ejecucion:%f", time_spent);
	
	return 0;
}

long sumaCuadrados(){
	int i;
	long acu;
	
	acu = 0;
	for(i=1;i<=100;i++){
		acu = acu + pow(i,2);
	}
	
	return acu;
	
}

long cuadradoSuma(){
	int i;
	long acu;
	
	acu = 0;
	for(i=1;i<=100;i++){
		acu = acu + i;
	}
	
	acu = pow(acu,2);
	return acu;
}
